packages/shared/src/contracts/agents/global-allow-list.ts

```diff
@@ -0,0 +1,56 @@
+import { ethers } from 'ethers';
+
+import { Domain } from '../../porter';
+import { ChecksumAddress } from '../../types';
+import { GlobalAllowList, GlobalAllowList__factory } from '../ethers-typechain';
+import { DEFAULT_WAIT_N_CONFIRMATIONS, getContract } from '../registry';
+
+export class GlobalAllowListAgent {
+  public static async registerEncrypters(
```

Comment:
Doesn't have to be this PR, but you may want to add a follow-up issue to add the ability to check if an encrypter is already registered i.e. `isAddressAuthorized(ritualid, address)` I think the contract method is called.

Reply:
You could also use it to check if the encrypter is already authorized before performing the authorization below.

Reply:
These two methods are [exposed by `taco`](https://github.com/nucypher/nucypher-ts/pull/314/files#diff-94274562e7c2dacfa0fd63a28f2d134b6c4b4a918761053b83b816b6a065adcbR109-R116). I think the latter (registering encrypters) could use the former under the hood.